txn_writer.go

```diff
@@ -436,43 +437,43 @@ fail:
 	return failerr
 }
 
-func (sw *SafeWriter) PrintPreparedActions() error {
+func (sw *SafeWriter) PrintPreparedActions(stdout *log.Logger) error {
```

Comment:
nit: let's name the local var `output` instead of `stdout`. Even if we always (currently) end up attaching stdout here, I don't think we need to suggest that via the var name.